pkg/diagnose/runner.go

```diff
@@ -400,3 +404,14 @@ func RunStdOut(w io.Writer, diagCfg diagnosis.Config, senderManager sender.Diagn
 
 	return nil
 }
+
+func getSuites(diagCfg diagnosis.Config, senderManager sender.DiagnoseSenderManager, collector optional.Option[collector.Component]) []diagnosis.Suite {
+	catalog := diagnosis.NewCatalog()
+
+	catalog.Register("check-datadog", func() []diagnosis.Diagnosis { return getDiagnose(diagCfg, senderManager, collector) })
+	catalog.Register("connectivity-datadog-core-endpoints", func() []diagnosis.Diagnosis { return connectivity.Diagnose(diagCfg) })
+	catalog.Register("connectivity-datadog-autodiscovery", connectivity.DiagnoseMetadataAutodiscoveryConnectivity)
+	catalog.Register("connectivity-datadog-event-platform", eventplatformimpl.Diagnose)
+
+	return catalog.GetSuites()
+}
```

Comment:
Interesting approach but I was thinking of distributed registration. And in part it is the reason I have passed diagnose config to every run-time suite specific diagnose execution.

Reply:
> I was thinking of distributed registration
Can you elaborate what you had in mind?

Reply:
Diagnose package had been envisioned as an oblivious catalog which will simply execute whatever any part of the agent is interested in self-diagnostics. It has no preconceived nothing of what these will be. It maybe runtime or build tag dependent. By moving explicit registration into diagnose package directly make things easier code wise but "breaks" the intent of more dynamic diagnostic system. How how other providers for status or inventory work, BTW?

Reply:
per discussion we will move the part into further PR.